proto/substrait/algebra.proto

```diff
@@ -340,6 +340,33 @@ message ExchangeRel {
   }
 }
 
+// Duplicates records, possibly switching output expressions between each duplicate.
+// Default output is all of the fields declared + one field that int64 type that returns the
+// duplicate_id which is a zero-index ordinal of which duplicate of the original record this
+// corresponds to.
+message ExpandRel {
+  RelCommon common = 1;
+  Rel input = 2;
+  repeated ExpandField fields = 4;
+
+  message ExpandField {
+    oneof field_type {
+      // Field that switches output based on which duplicate_id we're outputting
+      SwitchingField switching_field = 2;
+
+      // Field that outputs the same value no matter which duplicate_id we're on.
+      Expression consistent_field = 3;
+    }
+  }
+
+  message SwitchingField {
+    // Array that contains the expression to output per duplicate_id
+    // must have the same number of expressions as each input record
+    // all expressions must be of the same type a
```

Comment:
```suggestion
    // all expressions within a switching field be the same type class but can differ in nullability.
    // this column will be nullable if any of the expressions are nullable.
```

This one is debatable.  If we want them to be exactly the same then the expressions can be something like:

```
cast(field_ref("country"), string(nullable: true))
literal(null, type: string(nullable: true))
```

However, I think it is easier to allow each expression to have different nullability.  Then we can have:

```
field_ref("country"), string(nullable: true)
literal(null, type: string(nullable: true))
```

I'm fine either way.